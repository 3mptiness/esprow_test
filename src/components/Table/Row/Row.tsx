import React, { FC, memo } from 'react';
import { ChangeHandler, HeadersMap, RowType } from '../../../types';
import { Cell } from '../Cell';
import { DEFAULT_ROW_HEIGHT } from '../../../constants';

export const Row: FC<{
	id: string;
	index: number;
	row: RowType;
	headers: HeadersMap;
	onChange?: ChangeHandler;
	height?: number;
}> = memo((props) => {
	const {
		id,
		index,
		row,
		headers,
		onChange,
		height = DEFAULT_ROW_HEIGHT,
	} = props;

	return (
		<tr>
			{Array.from(headers).map(([title, type]) => (
				<Cell
					key={id + title}
					index={index}
					id={id}
					type={type}
					title={title}
					value={row[title]}
					onChange={onChange}
					height={height}
				/>
			))}
		</tr>
	);
});
