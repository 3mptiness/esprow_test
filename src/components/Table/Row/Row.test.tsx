import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { Row } from './Row';

describe('Row', () => {
	it('success row render', () => {
		expect(
			createRenderer().render(
				<Row
					headers={new Map([['testKey', 'input']])}
					row={{ testKey: 'test data', otherKey: 'other data' }}
					index={10}
					id="100500"
				/>
			)
		).toMatchSnapshot();
	});
});
