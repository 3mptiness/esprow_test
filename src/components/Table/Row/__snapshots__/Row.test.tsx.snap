// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row success row render 1`] = `
<tr>
  <Memo
    height={80}
    id="100500"
    index={10}
    title="testKey"
    type="input"
    value="test data"
  />
</tr>
`;
